assembly/Int/CompactInt.ts

```diff
@@ -43,7 +43,22 @@ export class CompactInt implements Codec {
 
         return bytesBuffer.bytes;
     }
-
+    /**
+     * @description Non-static constructor method used to populate defined properties of the model
+     * @param bytes SCALE encoded bytes
+     * @param index index to start decoding the bytes from
+     */
+    public populateFromBytes(bytes: u8[], index: i32 = 0): void{
+        assert(bytes.length - index > 0, "CompactInt: Empty bytes array provided");
+        const decodedData = Bytes.decodeCompactInt(bytes, index);
+        this.value = decodedData.value;
+        if (decodedData.value < 1 << 6) this.bitLength = BIT_LENGTH.INT_8;
+        else if (decodedData.value < 1 << 14) this.bitLength = BIT_LENGTH.INT_16;
+        else if (decodedData.value < 1 << 30) this.bitLength = BIT_LENGTH.INT_32;
+        else {
+            this.bitLength = BIT_LENGTH.INT_64;
+        }
```

Comment:
we could extract this logic into `_computeBitLength` internal function and reuse it in the constructor and here, just to remove the code duplication.